src/ape/contracts/base.py

```diff
@@ -849,13 +849,17 @@ def source_path(self) -> Optional[Path]:
         Returns the path to the local contract if determined that this container
         belongs to the active project.
         """
-        contract_name = self.contract_type.name
-        contract_container = self.project_manager._get_contract(contract_name)
+        if self.contract_type.name:
+            contract_name = self.contract_type.name
+        if self.project_manager._get_contract(contract_name):
```

Comment:
Can we achieve this same behavior without calling `self.project_manager._get_contract(contract_name)` twice?

Reply:
I think I broke it while trying to accomplish this. Still investigating 😅